ide/idetop.ml

```diff
@@ -208,32 +208,35 @@ let goals () =
   let doc = get_doc () in
   set_doc @@ Stm.finish ~doc;
   try
+    let oldp = Stm.get_prev_proof ~doc (Stm.get_current_state ~doc) in
     let newp = Proof_global.give_me_the_proof () in
-    if Proof_diffs.show_diffs () then begin
-      let oldp = Stm.get_prev_proof ~doc (Stm.get_current_state ~doc) in
+    match oldp with
+    | Some oldp when Proof_diffs.show_diffs () ->
```

Comment:
The easy case should be at the top of the match. It would be nice if the catch block were smaller too.
Maybe something like 
~~~ocaml
match Proof_global.give_me_the_proof () with 
  | exception Proof_global.NoCurrentProof -> None
  | Some newp -> 
    let dflt () = Some (export_pre_goals (Proof.map_structured_proof newp process_goal)) in
    if not show_diffs then dflt ()
    else match get_prev_proof ... with None -> dflt ()
    | Some oldp -> ... try .... with Diffs_failure -> dflt ()
~~~